src/root.zig

```diff
@@ -114,55 +114,81 @@ pub fn AlgorithmType(
 
         allocator: std.mem.Allocator,
 
+        initialised: bool,
+
         pub fn deinit(self: *Self) void {
-            self.m.deinit();
-            self.x.deinit();
-            self.m_skip.deinit();
-            self.allocator.free(self.role_bonus);
-            self.allocator.free(self.bonus_buffer);
-            self.allocator.free(self.first_match_buffer);
-            self.allocator.free(self.traceback_buffer);
+            self.deallocateMatrixAndBuffer();
             self.* = undefined;
         }
 
         pub fn init(
             allocator: std.mem.Allocator,
+        ) !Self {
+            // TODO: init all matrices and buffers
```

Comment:
This needs to be added. `init` should initialize, and the `initialised` field should be removed entirely, as it should always be initialised.